crates/paimon/src/spec/data_file.rs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::spec::schema::DataField;
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+
+/// Data type of a sequence of fields. A field consists of a field name, field type, and an optional
+/// description. The most specific type of a row of a table is a row type. In this case, each column
+/// of the row corresponds to the field of the row type that has the same ordinal position as the
+/// column. Compared to the SQL standard, an optional field description simplifies the handling with
+/// complex structures.
+///
+/// TODO: make RowType extends DataType.
+/// TODO: move me to a better place.
+pub struct RowType {
+    fields: Vec<DataField>,
+    nullable: bool,
+}
+
+impl Default for RowType {
+    fn default() -> Self {
+        Self {
+            fields: vec![],
+            nullable: true,
+        }
+    }
+}
+
+impl RowType {
+    pub const fn new(list: Vec<DataField>) -> Self {
+        Self {
+            fields: list,
+            nullable: true,
+        }
+    }
+    pub fn with_data_fields<F: Into<Vec<DataField>>>(list: F) -> Self {
+        Self {
+            fields: list.into(),
+            nullable: true,
+        }
+    }
+    pub fn with_nullable(mut self, nullable: bool) -> Self {
+        self.nullable = nullable;
+        self
+    }
+}
+
+pub const EMPTY_BINARY_ROW: BinaryRow = BinaryRow::new(0);
+
+// TODO: move me to a better place.
+#[derive(Debug, Eq, PartialEq)]
+pub struct BinaryRow {
+    arity: i32,
+    null_bits_size_in_bytes: i32,
+}
+
+impl BinaryRow {
+    pub const HEADER_SIZE_IN_BYTES: i32 = 8;
+    pub const fn cal_bit_set_width_in_bytes(arity: i32) -> i32 {
+        ((arity + 63 + Self::HEADER_SIZE_IN_BYTES) / 64) * 8
+    }
+    pub const fn cal_fix_part_size_in_bytes(arity: i32) -> i32 {
+        Self::cal_bit_set_width_in_bytes(arity) + 8 * arity
+    }
+    pub const fn new(arity: i32) -> Self {
+        Self {
+            arity,
+            null_bits_size_in_bytes: (arity + 7) / 8,
+        }
+    }
+}
+
+/// TODO: implement me.
+type SimpleStats = ();
+
+type Timestamp = u64;
```

Comment:
We can use [chrono::DateTime](https://docs.rs/chrono/latest/chrono/struct.DateTime.html) with UTC timezone.